renderdoc/driver/ihv/amd/amd_counters.cpp

```diff
@@ -110,13 +109,41 @@ bool AMDCounters::Init(ApiType apiType, void *pContext)
     return false;
   }
 
+  bool disableCounters = false;
+
+#ifndef RELEASE
+
+  if(apiType == ApiType::Dx12)
+  {
+    // Disable counters in DX12 Debug configuration
+    void *versionFunc = Process::GetFunctionAddress(module, "GPA_GetVersion");
```

Comment:
I'm not sure I understand this check - is it intended to detect GPA 3.0? This code seems like it will be incompatible with the 3.0 preview anyway so I'm not sure if that's necessary.

Reply:
Yes Baldur, you are right! This is just to detect the version 3.0. We are adding this entry point in 3.1. This check is to make sure the backward compatibility between GPA releases based out  of v3.0.

Reply:
Oh right I see, I thought this was a new function in 3.0.

In that case it's trickier then. I still don't want to check only on development/release builds since really what you want to look for is if the debug layer is enabled, but I see your point that it's best to have the version checking here. I think then you just need to pass in a flag about whether the debug layer is enabled or not - unless there's a way I've missed to query from D3D whether the debug layer is on.

Reply:
Now, I added a flag in RenderDoc class to keep the status of DX12 Debug layer and using it to toggle the AMD hardware counter behavior in the debug builds. Let me know if you have any other preference to handle this flag.

Reply:
Sorry, but I don't want to have API-specific details like DX12 debug layer status in the core code. This flag should stay within the D3D12 project - I'd suggest like before that you instead pass this flag to the `WrappedID3D12Device` constructor, then pass it again when constructing the `AMDCounters` object.